llvm/lib/Transforms/Scalar/InferAddressSpaces.cpp

```diff
@@ -989,7 +986,8 @@ bool InferAddressSpacesImpl::updateAddressSpace(
           OperandAS = PtrOperand->getType()->getPointerAddressSpace();
           if (OperandAS == FlatAddrSpace) {
             // Check AC for assumption dominating V.
-            unsigned AS = getPredicatedAddrSpace(V, PtrOperand);
+            unsigned AS =
+                getPredicatedAddrSpace(*PtrOperand, &cast<Instruction>(V));
```

Comment:
Can we always cast to `Instruction`? E.g. what if `V` is an `Argument`?

Reply:
This is currently only used from the result of getPointerOperands, so it must at least be Operator. It could maybe be  a constantexpr, but I'm not sure it's possible

Reply:
it's still a dyn_cast->cast change